src/network/networkd-dhcp-server.c

```diff
@@ -383,6 +385,17 @@ int dhcp4_server_configure(Link *link) {
                         return log_link_error_errno(link, r, "Failed to set DHCPv4 option: %m");
         }
 
+        ORDERED_HASHMAP_FOREACH(static_lease, link->network->dhcp_static_leases_by_section) {
+                /* Static lease with an empty or omitted address is a valid entry,
+                * it removes any static lease with the specified mac address. */
+                if (sd_dhcp_static_lease_check_empty_address(static_lease->static_lease) == 1)
+                        r = sd_dhcp_server_remove_static_lease(link->dhcp_server, static_lease->static_lease);
+                else
+                        r = sd_dhcp_server_add_static_lease(link->dhcp_server, static_lease->static_lease);
+                if (r < 0)
+                        return log_link_error_errno(link, r, "Failed to set DHCPv4 static lease for DHCP server: %m");
+        }
```

Comment:
i don't get this part. Why isn't the DHCP static list initialized afresh whenever a link is configured? what's this stuff about "removing" a lease here? don't get this.

Reply:
The list of DHCP static list is configured in the config file and passed on to the DHCP server which is being configured. The addresses are being removed if a ```[DHCPServerStaticLease]````section is being overwritten with a section that has the same MAC address, but an empty IP address. This is a feature that was specifically requested. I hope I cleared some stuff out.